allennlp/models/encoder_decoders/wikitables/wikitables_semantic_parser.py

```diff
@@ -136,12 +155,66 @@ def __init__(self,
                                                    mixture_feedforward=mixture_feedforward,
                                                    dropout=dropout)
 
+        self._executor_process: subprocess.Popen = None
+        self._create_sempre_executor()
+
+    def _create_sempre_executor(self) -> None:
```

Comment:
Seems like it would be better if you implemented this as a `Metric`? We have other very specific metrics, like the coref stats there already, and it would be a good example for how to have a metric which does complex interactions with other code.